### ISSUE TITLE
textAngular stops working on Firefox 42.0 when blurring/focusing

### ISSUE THREAD

**Issue by M4tiz** [fraywing/textAngular#993]
When on Firefox 42.0, textAngular randomly stops responding to any keyboard input when you repeatedly blur and focus the input.

Steps to reproduce:
1. Open http://textangular.com/ in Firefox
2. Remove everything from the example input
3. Blur the input
4. Focus the input
5. Try to type something

Reproduction rate is 100%. Bug occurs very often on Ubuntu, but works on Windows as well.

I think this occurs only on the newest 42.0 Firefox version.

From what I managed to deduce while debugging the source code, it may be related to the *_blankTest* function. It returns true when the new model value has any HTML tags and the model value in the 
*_setViewValue* function is not updated.

Furthermore, in this scenario, the value passed to the *_blankTest* function is given by *_compileHtml* function. Which returns `<br>` at the end of our input value ( `element[0].innerHTML` ).

**Comment by gbdlin** [comment]
I'm confirming error, input is not working after removing all text and blurring. When re-focused, input is no longer responding to any keystrokes.

If input was empty by default, focusing and blurring doesn't change anything. You must type something and remove it to reproduce error.

**Comment by fgodet** [comment]
I also have this issue, using Frefox 43.0.
Same thing with textAngular 1.4.1 but with 1.3.11 it seems to work fine ... I was not able to reproduce the bug ...

**Comment by msibee** [comment]
can you show jsbin code snippet?

**Comment by dlehmann** [comment]
same problem with firefox 45.0.1

**Comment by tonyfa** [comment]
Any workaround for this or can anyone suggest a good free alternative?

**Comment by tonyfa** [comment]
I just swapped it out for this

https://github.com/TerryMooreII/angular-wysiwyg

**Comment by tonyfa** [comment]
I used this workaround if they are using firefox I don't set the focus otherwise I do

```
if (navigator.userAgent.indexOf("Firefox") < 1) {
                                    $timeout(function () {
                                        angular.element('text-angular[name=txtQuestionId]').find('*[id*=taText]').focus();
                                    }, 1000);
                                }
```

**Comment by pagas** [comment]
It is not enough to trigger focus on angular text, because it places cursor outside of p tag, then angular text automatically adds line breaks and more p tags which messes everything up. So, we insert cursor inside p tag. And this is just FF.

Here is code snippet I used [](http://jsfiddle.net/vXnCM/5527/ )
`var containerNode;
var elementToSelect;

$(function() {
  containerNode = $("#editable")[0];
  elementToSelect = $('p', containerNode).first()[0];
  
	focustAtBeginning();
})
  
function focustAtBeginning() {
	console.log('do it');
  putCarotInFirstNone(containerNode, elementToSelect)
}

function putCarotInFirstNone(containerNode, selectNone) {
  var range = document.createRange();
  var sel = window.getSelection();

  range.setStart(selectNone, 0);

  range.collapse(true);
  sel.removeAllRanges();

  sel.addRange(range);
  containerNode.focus();
}





`

**Comment by JoelParke** [comment]
this is fixed 1.5.6+  Please reopen if still an issue.
